partner_relations/models/res_partner_relation.py

```diff
@@ -48,44 +48,44 @@ def create(self, vals):
             vals['left_partner_id'] = context.get('active_id')
         return super(ResPartnerRelation, self).create(vals)
 
-    @api.one
     @api.constrains('date_start', 'date_end')
     def _check_dates(self):
         """End date should not be before start date, if not filled
 
         :raises exceptions.Warning: When constraint is violated
         """
+        self.ensure_one()
```

Comment:
`ensure_one` simply raises if there's no or more than one record. If you actually want to get rid of the `@api.one` decorators, you need to rewrite the function to cope with recordsets. Especially constrains really need to be able to handle an arbitrary number of records. But the decorator has been de-deprecated to my knowledge (citation needed), so I'd ignore those warnings.